octree/basic_utils.go

```diff
@@ -92,3 +92,34 @@ func (octree *basicOctree) checkPointPlacement(p r3.Vector) bool {
 		(math.Abs(octree.center.Y-p.Y) <= octree.sideLength/2.) &&
 		(math.Abs(octree.center.Z-p.Z) <= octree.sideLength/2.))
 }
+
+// helperIterateRecursive is a helper function for iterating through a basic octree. If an internal
+// node is found it will be called recursively on each child, updating the idx to correspond to the
+// number of data points in each child. By doing so, each leaf node containing data will have a unique
+// idx value. If the currentBatch number matches the idx%numBatches then the specified function will
+// be performed on the point. If the function returns false, the iteration will end.
+func (octree *basicOctree) helperIterateRecursive(numBatches, currentBatch int, idx uint,
```

Comment:
+1 for avoiding recursion, rather than setting a limit to the depth of octrees. 

I foresee that if we chose the latter, we'd have to refactor the code eventually anyway. Rather than users running into issues due to this choice and us having to rush to fix it, we could refactor now and avoid a stressful situation later on. Thoughts?

Reply:
I'm on the fence. I definitely see the argument for taking the extra time to get the eventual implementation we want now. On the other hand, we don't yet have any uses of octrees, so we don't know if this will be common or would usually indicate a malicious input. We at least need to do one or the other (add a limit or avoid recursion), so we're not susceptible to attack.

Reply:
Leaving this comment open in case @kkufieta has more to say.